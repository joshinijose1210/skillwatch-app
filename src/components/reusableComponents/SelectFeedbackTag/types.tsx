export interface SelectFeedbackProps {
    feedbackTagValue: string | undefined;
    setFeedbackTagValue: React.Dispatch<React.SetStateAction<string>>;
    dataTestId?: string;
    disabled?: boolean;
}
