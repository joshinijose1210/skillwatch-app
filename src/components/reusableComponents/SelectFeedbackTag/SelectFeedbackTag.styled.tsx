import styled from 'styled-components';

export const SelectFeedbackTagWrapper = styled.div`
    min-width: 500px;
`;
