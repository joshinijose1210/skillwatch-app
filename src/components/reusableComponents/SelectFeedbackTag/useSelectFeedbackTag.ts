import { useGetTagsListQuery } from '@slice/services';

export const useSelectFeedbacktag = () => {
    const { data: tags, isSuccess: isTagsListReceived } = useGetTagsListQuery({});

    const tagsList = isTagsListReceived
        ? [
              ...tags.map((tag: any) => {
                  return { value: tag.feedbackTypeId, label: tag.feedbackType };
              })
          ]
        : [];

    return { tagsList };
};
