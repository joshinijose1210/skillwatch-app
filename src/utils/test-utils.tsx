import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
