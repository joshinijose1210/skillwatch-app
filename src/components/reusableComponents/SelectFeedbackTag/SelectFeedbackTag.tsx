import { StyledSingleSelect } from '@common';
import { SelectFeedbackTagWrapper } from './SelectFeedbackTag.styled';
import { SelectFeedbackProps } from './types';
import { useSelectFeedbacktag } from './useSelectFeedbackTag';
import TooltipDropdown from '@components/reusableComponents/TooltipDropdown';
import { OptionsType } from '@components/reusableComponents/TooltipDropdown/types';

export const SelectFeedbackTag = ({ feedbackTagValue, setFeedbackTagValue, disabled = false, dataTestId }: SelectFeedbackProps) => {
    const { tagsList } = useSelectFeedbacktag();

    return (
        <SelectFeedbackTagWrapper>
            <StyledSingleSelect
                disabled={disabled}
                options={tagsList}
                value={feedbackTagValue}
                onChange={val => val && setFeedbackTagValue(val)}
                label={'Feedback Type'}
                placeholder={'Select Feedback Type'}
                variant="outlined"
                size="M"
                minWidth="100%"
                data-testid={dataTestId}
            />
            <TooltipDropdown
                dataIds={[`feedback type-input`]}
                values={feedbackTagValue && tagsList.length ? tagsList.filter((item: OptionsType) => feedbackTagValue === item.value) : []}
            />
        </SelectFeedbackTagWrapper>
    );
};
