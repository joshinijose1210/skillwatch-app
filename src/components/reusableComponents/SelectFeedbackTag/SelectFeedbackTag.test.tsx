import { RTKRender } from '@utils/test-utils';
import { SelectFeedbackTag } from './SelectFeedbackTag';

describe('SelectFeedbackTag component', () => {
    it('should take snapshot', () => {
        const { container } = RTKRender(<SelectFeedbackTag feedbackTagValue={''} setFeedbackTagValue={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });
});
