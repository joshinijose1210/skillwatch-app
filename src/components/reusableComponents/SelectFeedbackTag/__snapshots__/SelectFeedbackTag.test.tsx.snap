// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFeedbackTag component should take snapshot 1`] = `
<div>
  <div
    class="sc-ayn6ri-0 boWEAe"
  >
    <div
      class="sc-1hcsztp-0 fjPkSb sc-tkiih6-13 cecHDR"
      id="feedback type-wrapper"
      minwidth="100%"
      variant="outlined"
    >
      <div
        class="sc-17d4s39-0 khebHA"
        id="feedback type-input-wrapper"
        minwidth="100%"
      >
        <div
          class="sc-19px6zp-0 BrSGZ"
          filled="[object Object]"
          fusion="[object Object]"
          minrows="1"
          outlined="[object Object]"
          textvariant="[object Object]"
          variant="outlined"
        >
          <div
            class="sc-lhr232-0 btNOKw"
            variant="outlined"
          >
            <input
              aria-describedby="feedback type-helper-text"
              autocomplete="off"
              class="sc-hb99wz-0 jaEbhp"
              data-tooltip-id="feedback type-input"
              errortext=""
              id="feedback type-input"
              inputmode="none"
              inputsize="M"
              placeholder="Select Feedback Type"
              type="text"
              value=""
              variant="outlined"
            />
            <label
              class="sc-1du51u0-0 eMZrST"
              for="feedback type-input"
              inputwidth="0"
              variant="outlined"
            >
              Feedback Type
            </label>
          </div>
          <svg
            class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </div>
        
      </div>
    </div>
    
  </div>
</div>
`;
